src/Microsoft.Health.Fhir.SqlServer/Features/Storage/Registry/SqlServerSearchParameterStatusDataStore.cs

```diff
@@ -0,0 +1,115 @@
+using Microsoft.Health.SqlServer.Features.Storage;
+using SqlDataReader = Microsoft.Data.SqlClient.SqlDataReader;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Storage.Registry
+{
+    internal class SqlServerSearchParameterStatusDataStore : ISearchParameterStatusDataStore
+    {
+        private readonly Func<IScoped<SqlConnectionWrapperFactory>> _scopedSqlConnectionWrapperFactory;
+        private readonly VLatest.UpsertSearchParamsTvpGenerator<List<ResourceSearchParameterStatus>> _updateSearchParamsTvpGenerator;
+        private readonly ISearchParameterStatusDataStore _filebasedSearchParameterStatusDataStore;
+        private readonly SchemaInformation _schemaInformation;
+
+        public SqlServerSearchParameterStatusDataStore(
+            Func<IScoped<SqlConnectionWrapperFactory>> scopedSqlConnectionWrapperFactory,
+            VLatest.UpsertSearchParamsTvpGenerator<List<ResourceSearchParameterStatus>> updateSearchParamsTvpGenerator,
+            FilebasedSearchParameterStatusDataStore.Resolver filebasedRegistry,
+            SchemaInformation schemaInformation)
+        {
+            EnsureArg.IsNotNull(scopedSqlConnectionWrapperFactory, nameof(scopedSqlConnectionWrapperFactory));
+            EnsureArg.IsNotNull(updateSearchParamsTvpGenerator, nameof(updateSearchParamsTvpGenerator));
+            EnsureArg.IsNotNull(filebasedRegistry, nameof(filebasedRegistry));
+            EnsureArg.IsNotNull(schemaInformation, nameof(schemaInformation));
+
+            _scopedSqlConnectionWrapperFactory = scopedSqlConnectionWrapperFactory;
+            _updateSearchParamsTvpGenerator = updateSearchParamsTvpGenerator;
+            _filebasedSearchParameterStatusDataStore = filebasedRegistry.Invoke();
+            _schemaInformation = schemaInformation;
+        }
+
+        public async Task<IReadOnlyCollection<ResourceSearchParameterStatus>> GetSearchParameterStatuses()
+        {
+            // If the search parameter table in SQL does not yet contain status columns
+            if (_schemaInformation.Current < SchemaVersionConstants.SearchParameterStatusSchemaVersion)
+            {
+                // Get status information from file.
+                return await _filebasedSearchParameterStatusDataStore.GetSearchParameterStatuses();
+            }
+
+            using (IScoped<SqlConnectionWrapperFactory> scopedSqlConnectionWrapperFactory = _scopedSqlConnectionWrapperFactory())
+            using (SqlConnectionWrapper sqlConnectionWrapper = await scopedSqlConnectionWrapperFactory.Value.ObtainSqlConnectionWrapperAsync(CancellationToken.None, true))
+            using (SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateSqlCommand())
+            {
+                VLatest.GetSearchParamStatuses.PopulateCommand(sqlCommandWrapper);
+
+                var parameterStatuses = new List<ResourceSearchParameterStatus>();
+
+                // TODO: Make cancellation token an input.
```

Comment:
Is ithis still a TODO? I see that in below statement we are passing CancellationToken

Reply:
Ah, I see how this is unclear. By this, I intended to indicate that a cancellation token should be an input to `GetSearchParameterStatuses()` and `UpsertStatuses()`. I moved the comment to be above those methods in 229f90c so that it's more clear.

I remember creating a user story for this at one point. I can't seem to find it anymore, so I created a new one: [#AB77269](https://microsofthealth.visualstudio.com/Health/_workitems/edit/77269)